[17.0][MIG] sale_loyalty_partner_applicability: Migration to 17.0

arantxa-s73 opened the pull request:
Module migrated to version 17.0

Depends on:

- [ loyalty_partner_applicability ] https://github.com/OCA/sale-promotion/pull/225

pilarvargas-tecnativa committed:
[ADD] sale_loyalty_partner_applicability: New module

TT45413

Commit:
[UPD] Update sale_loyalty_partner_applicability.pot

OCA-git-bot committed:
[BOT] post-merge updates

pilarvargas-tecnativa committed:
[FIX] sale_loyalty_partner_applicability: correct domain evaluation

TT44347

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (4 of 4 strings)

Translation: sale-promotion-16.0/sale-promotion-16.0-sale_loyalty_partner_applicability
Translate-URL: https://translation.odoo-community.org/projects/sale-promotion-16-0/sale-promotion-16-0-sale_loyalty_partner_applicability/it/

pedrobaeza committed:
[IMP] sale_loyalty_partner_applicability: Don't evaluate domain if not set

This way, we can avoid a crash on incorrect input data.

OCA-git-bot committed:
[BOT] post-merge updates

pilarvargas-tecnativa committed:
[FIX] sale_loyalty_partner_applicability: fix test_sale_loyalty_limit

When the "next_order_coupon" promotion that generates coupons meets the
conditions of a sales order, it may cause an error if the appropriate
coupon code is not applied. In these tests, coupons from other sales
orders are being generated for this promotion, so it is advisable to
retrieve the exact coupon to be applied by filtering by the sales order
that generated it in order to avoid conflicts.

TT47449

OCA-git-bot committed:
[BOT] post-merge updates

arantxa-s73 committed:
[IMP] sale_loyalty_partner_applicability: pre-commit auto fixes

arantxa-s73 committed:
[MIG] sale_loyalty_partner_applicability: Migration to 17.0

rousseldenis commented:
@arantxa-s73 Don't hesitate to put in this PR's description the link to the depending PR's in order to easy review. Thanks

rousseldenis commented:
/ocabot migration sale_loyalty_partner_applicability

lmignon committed:
[IMP] loyalty_partner_applicability: Fix partner applicability

Before this commit, the partner applicability was not working according
to the addon description. For programs with multiple rules, rules defined
for specific partners were being applied to any partner as long as the
partner had a rule defined for them. The same applied to the coupon, where
a coupon defined for a specific partner could be used by any partner as long
the partner had a ruele defined for them in the same program where
the coupon was defined.

This commit fixes the issue by ensuring that the partner applicability
is only checked for the partner that the rule or coupon is defined for.
It's done by removing the rules not applicable to the the partner linked
to the sale order when the system is requesting the rules to be applied
to the products in the sale order. When a coupon is applied we also checke
if the partner is linked to the rule at the origin of the coupon. Or to
a rule linked to the loyalty card program.

lmignon committed:
[IMP] loyalty_partner_applicability: Fix partner applicability

Before this commit, the partner applicability was not working according
to the addon description. For programs with multiple rules, rules defined
for specific partners were being applied to any partner as long as the
partner had a rule defined for them. The same applied to the coupon, where
a coupon defined for a specific partner could be used by any partner as long
the partner had a ruele defined for them in the same program where
the coupon was defined.

This commit fixes the issue by ensuring that the partner applicability
is only checked for the partner that the rule or coupon is defined for.
It's done by removing the rules not applicable to the the partner linked
to the sale order when the system is requesting the rules to be applied
to the products in the sale order. When a coupon is applied we also checke
if the partner is linked to the rule at the origin of the coupon. Or to
a rule linked to the loyalty card program.

lmignon committed:
[IMP] loyalty_partner_applicability: Improve the applicability of loyalty programs to partners

This change introduces a new mixin to allow the definition of the applicability
of a loyalty program to a partner. This mixin is used in the loyalty program model
to define the applicability of the program to partners. The mixin is also used
in the loyalty rule model to define the applicability of the rule to partners.

With this change, the applicability of a loyalty program to a partner can be
now defined at the program and rule level.

lmignon committed:
[IMP] loyalty_partner_applicability: Improve the applicability of loyalty programs to partners

This change introduces a new mixin to allow the definition of the applicability
of a loyalty program to a partner. This mixin is used in the loyalty program model
to define the applicability of the program to partners. The mixin is also used
in the loyalty rule model to define the applicability of the rule to partners.

With this change, the applicability of a loyalty program to a partner can be
now defined at the program and rule level.

lmignon committed:
[IMP] loyalty_mass_mailing: Adapt the module to the new way of applying loyalty programs to partners

The partner applicability of a loyalty rule can now be defined at the level of the loyalty program
in addition to the level of the loyalty rule. This commit adapts the module to this new way of
defining the partner applicability of a loyalty program when computing the domain to use for
the mass mailing of a loyalty program.

lmignon reviewed (approved):
LGTM (code review only with full diff with the 16.0 branch)

pilarvargas-tecnativa commented:
Please review the changes made in this PR, which has already been merged.
https://github.com/OCA/sale-promotion/pull/265

arantxa-s73 commented:
@pilarvargas-tecnativa the changes are already in the pr

pilarvargas-tecnativa commented:
> @pilarvargas-tecnativa the changes are already in the pr

There are new changes and new commits that have not yet been incorporated.

arantxa-s73 commented:
@pilarvargas-tecnativa done

Reyes4711-S73 reviewed (approved):
LGTM

Reyes4711-S73 commented:
@OCA/crm-sales-marketing-maintainers Please, can you merge this PR?
We can't improve more the codecov lines. We tested all lines of this module yet

pedrobaeza commented:
/ocabot merge nobump
